airbyte-integrations/connectors/source-redshift/src/test/java/io/airbyte/integrations/source/redshift/RedshiftSpecTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.redshift;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.airbyte.commons.io.IOs;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.resources.MoreResources;
+import io.airbyte.protocol.models.ConnectorSpecification;
+import io.airbyte.validation.json.JsonSchemaValidator;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests that the Redshift spec passes JsonSchema validation. While this may seem like overkill, we
+ * are doing it because there are some gotchas in correctly configuring the oneOf.
+ */
+public class RedshiftSpecTest {
+
+  private static final String CONFIGURATION = """
+                                              {
+                                                "host": "localhost",
+                                                "port": 1521,
+                                                "database": "db",
+                                                "schemas": [
+                                                  "public"
+                                                ],
+                                                "username": "redshift",
+                                                "password": "password",
+                                                "jdbc_url_params": "property1=pValue1&property2=pValue2"
+                                              }
+                                              """;
+
+
+  private static JsonNode schema;
+  private static JsonSchemaValidator validator;
+
+  @BeforeAll
+  static void init() throws IOException {
+    final String spec = MoreResources.readResource("spec.json");
+    final File schemaFile = IOs.writeFile(Files.createTempDirectory(Path.of("/tmp"), "pg-spec-test"), "schema.json", spec).toFile();
+    schema = JsonSchemaValidator.getSchema(schemaFile).get("connectionSpecification");
+    validator = new JsonSchemaValidator();
+  }
+
+  @Test
+  void testHostMissing() {
+    final JsonNode config = Jsons.deserialize(CONFIGURATION);
```

Comment:
It makes sense to have a get method for this operation and call it in the BeforeEach